crates/holochain_state/src/query/chain_head.rs

```diff
@@ -24,12 +24,11 @@ impl Query for ChainHeadQuery {
 
     fn query(&self) -> String {
         "
-        SELECT blob, hash FROM (
-            SELECT Action.blob, Action.hash, MAX(action.seq)
```

Comment:
This seemed like a problem to me. The MAX wasn't covered by tests and doesn't appear to be referring to the `Action` table even. I removed it and nothing broke. Because `MAX` is free to pick a different row to the values returned for blob and hash I'm pretty sure it wasn't doing anything. I've changed the query to be explicit about order and only return the actual chain head from the database's perspective.

Reply:
Thinking about the sort+limit last night, I think any cost incurred by doing that is a trade-off against having to deserialize all the rows that would come back in the existing query. This is more likely to be a performance improvement than cost

Reply:
There *are* indexes on [Action.hash](https://github.com/holochain/holochain/blob/3918b35bd39d038df40029c0402460894a1ea42a/crates/holochain_sqlite/src/sql/cell/schema/1.sql#L31), [DhtOp.action_hash](https://github.com/holochain/holochain/blob/3918b35bd39d038df40029c0402460894a1ea42a/crates/holochain_sqlite/src/sql/cell/schema/1.sql#L141) and [Action.seq](https://github.com/holochain/holochain/blob/3918b35bd39d038df40029c0402460894a1ea42a/crates/holochain_sqlite/src/sql/cell/schema/1.sql#L80).

I guess that's the best we can do without getting into runtime query plan analysis.

:+1: